src/cpp/rtps/builtin/discovery/participant/PDPServer2.cpp

```diff
@@ -1269,8 +1274,10 @@ bool PDPServer2::remove_change_from_history_nts(
 {
     for (auto chit = history->changesRbegin(); chit != history->changesRend(); chit++)
     {
-        if (change->instanceHandle == (*chit)->instanceHandle &&
-                change->write_params.sample_identity() == (*chit)->write_params.sample_identity())
+        // We compare by pointer because we maintain the same pointer everywhere and it is unique
+        // We cannot compare by cache info because there is no distinct attributes for the same change arrived
+        //  from different servers, and one of them could be in the history while the other arrive to ddb
```

Comment:
```suggestion
        // We cannot compare by cache info because there is no distinct attributes when the same change arrives
        // from different servers. One of them could be in the history while the other arrives to ddb
```